.github/workflows/test.yml

```diff
@@ -75,32 +76,32 @@ jobs:
         run: |
           pip install mypy black
 
-      - name: Restore CPM cache
-        env:
-          cache-name: cpm-cache-0
-        id: cpm-cache-restore
-        uses: actions/cache/restore@6849a6489940f00c2f30c0fb92c6274307ccb58a # v4.1.2
-        with:
-          path: ~/cpm-cache
-          key: ${{runner.os}}-${{env.cache-name}}-${{ hashFiles('**/CMakeLists.txt', 'cmake/**') }}
-          restore-keys: |
-            ${{runner.os}}-${{env.cache-name}}-
-
       - name: Configure cmake for lib
         env:
           CC: "/usr/lib/llvm-${{env.TARGET_LLVM_VERSION}}/bin/clang"
           CXX: "/usr/lib/llvm-${{env.TARGET_LLVM_VERSION}}/bin/clang++"
           PR_TARGET_BRANCH: ${{ steps.target_branch.outputs.branch }}
         working-directory: ${{github.workspace}}/test/library
-        run: cmake -Bbuild -DCPM_SOURCE_CACHE=~/cpm-cache
+        run: cmake -Bbuild
 
       - name: Check lib quality
         working-directory: ${{github.workspace}}/test/library
         run: cmake --build build -t ci-quality
 
       - name: Build lib docs
         working-directory: ${{github.workspace}}/test/library
-        run: cmake --build build -t docs
+        run: test $(cmake --build build -v -t docs | grep -c ERROR) == 0
```

Comment:
**note**: This change means that if image generation fails, this step fails. Before this change, image generation could fail and cmake would still succeed...